noir-projects/noir-protocol-circuits/crates/public-kernel-lib/src/components/enqueued_call_data_validator.nr

```diff
@@ -86,14 +85,6 @@ impl EnqueuedCallDataValidator {
         );
     }
 
-    fn validate_counters(self, previous_kernel: PublicKernelCircuitPublicInputs) {
-        assert_eq(
-            self.enqueued_call.data.start_side_effect_counter,
-            previous_kernel.end_side_effect_counter + 1,
-            "enqueued call must start from the end counter of the previous call",
-        );
-    }
-
     // Validates that the start gas injected into the vm circuit matches the remaining gas.
```

Comment:
This constraint was failing because when an enqueued call had no side effects, it would start with `start == prev.end`. I was trying to figure out the proper way to modify this constraint or modify the AVM/tracing when I realized.... We're going to delete this anyway since side effect counters shouldn't be a critical part of the protocol with everything in the AVM.